web/static/js/reducers/questionnaire.js

```diff
@@ -395,20 +440,27 @@ const validateSteps = (path, steps, context) => {
 }
 
 const validateStep = (path, step, context) => {
-  if (context.sms && isBlank(step.prompt['en'].sms)) {
+  if (context.sms &&
```

Comment:
I had to make all this guards super defensive because of the broken invariants resulting from the way `addLanguage` and `setDefaultLanguage` are implemented. The way to get away from this IMO is to provide more robust implementations to those actions so that these guards can be relaxed.